test/e2e/dynamic_provisioning_test.go

```diff
@@ -37,21 +37,24 @@ import (
 
 var _ = ginkgo.Describe("Dynamic Provisioning", func() {
 	t := dynamicProvisioningTestSuite{}
-
-	ginkgo.Context("[single-az]", func() {
-		t.defineTests(false)
-	})
-
-	ginkgo.Context("[multi-az]", func() {
-		t.defineTests(true)
-	})
+	schedulers := getListOfSchedulers()
+	for _, scheduler := range schedulers {
```

Comment:
Instead of unconditionally running every test case for all schedulers, it would be good to handle the scheduler name as a context. That way, we could limit the run to one case vs. the other for comparison and efficiency. It would allow us to more readily identify when an e2e failure is due to the scheduler extender by having separate e2e test pipelines.

Reply:
I think ginkgo lets you dynamically generate tests for each test case but I am not sure it lets you dynamically generate the context: https://onsi.github.io/ginkgo/#patterns-for-dynamically-generating-tests

I'll try adding the scheduler to the context dynamically to see if ginkgo discovers all the test cases. But even if this doesn't work, there is an environment variable that specifies whether to run the default or all the schedulers. We could change that var to run either the default or the scheduler extender. I just thought that if we have one testing on github/canary it would be good to have an option that would test all the schedulers by default for each PR.

Reply:
btw, the scheduler name is added to each test case, so with the correct regexp configuration for running the tests we can only run the tests for the specified scheduler.